intl_translation_format/README.md

```diff
@@ -56,6 +56,169 @@ pub run intl_translation_format:generate
+ } 
+ ```
+ 2. It is a text-based format
+ 3. It contains one locale translation per file.
+ 4. We will use the ICU standard for plurals, genders, and variables.
+
+
+2. Start coding
+
+ - Create a new dart package and add intl_translation_format as a dependency.
+
+- Choose what `TranslationFormat` class to extend. In our case, our format will be monolingual and text-based so we will extend `MonolingualFormat`. You can see the implementation of other classes in other packages.
+
+```dart 
+class JsonFormat extends MonoLingualFormat {
+
+  @override
+  String get fileExtension => 'json';
+  ...
+}  
+```
+The first thing will be to add the `fileExtension` in our case `JSON`. You can override the `fileExtensions` list param in case your format supports multiple extensions (eg. Xliff supports .xliff and xlf).
+
+3. Implement the method to generate a template.
+
+This method will generate a file in the desired format that will contain the messages from the template.
+
+```dart
+@override
+  String generateTemplateFile(
+    TranslationTemplate catalog,
+  ) {
+    final messages = catalog.messages;
+    var json = '{\n';
+    messages.forEach((key, value) {
+      // Expands the MainMessage into an icu string
+      final message = messageToIcuString(value);
+      json += '  "$key": "$message",\n';
+    });
+    if (messages.isNotEmpty) json = json.substring(0, json.length - 2) + '\n';
+    json += '}';
+    return json;
+  }
+```
+The param `catalog` contains the list of `MainMessage`s. These `MainMessages` can be expanded into a string with your transform function. For ICU messages there is an already built method `messageToIcuString` that expands the message into an ICU string.
+
+This method should return a String that contains the file content with the desired format.
+
+4. Implement the method to extract messages: 
+
+It will return `MessagesForLocale` object with the messages and an optional locale in case it is indicated inside the file.
+
+```dart
+@override
+  MessagesForLocale parseFile(
+    String content, {
+    MessageGeneration generation,
+  }) {
+    final json = jsonEncoder.decode(content) as Map<String, Object>;
+    final messages = <String, BasicTranslatedMessage>{};
+     // foldMap Creates a flat map with a set of keys per value.
+    foldMap<String>(json).forEach(
+      (keys, value) {
+        final unformattedKey =
+            keys.reduce((string1, string2) => '$string1 $string2');
+
+        final key = CaseFormat(Case.camelCase).format(unformattedKey);
+
+        final message = BasicTranslatedMessage(key, IcuMessage.fromIcu(value));
+        messages[key] = message;
+      },
+    );
+    return MessagesForLocale(messages);
+```
+
+The important part here is to parse the string message into a Message class. You can manually create a Message tree but for ICU there is already a default parser: `IcuMessage.fromIcu(value)`
```

Comment:
'manually parse the Message to create a syntax tree' ? Maybe mention the grammar?